### PR TITLE
setup: add PR and Issue templates

### PR DESCRIPTION
**Pull request by karansapolia**
fix: #15

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- [ ] Refactoring (no functional changes, no api changes)
+- [ ] Build related changes
+- [ ] CI related changes
+- [ ] Documentation content changes
+- [ ] Infrastructure changes
+- [ ] Dependency upgrade
+- [ ] Other... Please describe:
+
+## Issue Number: 
+#
```
Comment (rishabh1403) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe add Resolves: # too here ?

Comment (karansapolia) [reply .github/PULL_REQUEST_TEMPLATE.md]: Isn't it the same thing? I can change `## Issue Number: ` to `# Resolves issue number: # ` if that sounds better?

**Comment by karansapolia** [comment]
Wish to merge this as it is part of documentation and project structure only, the failing circleci tests font apply to this one. :slightly_smiling_face:

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (rishabh1403) [reply .github/PULL_REQUEST_TEMPLATE.md]: Resolves key word will auto close the issue , not sure if Resolves issue numbe will. Just double check once :)

**Pull request merged**